framework/generated/khronos_generators/khronos_base_generator.py

```diff
@@ -909,6 +912,25 @@ def genStruct(self, typeinfo, typename, alias):
                 typeinfo.elem.findall('.//member')
             ))
 
+            # Set the structure type value for this name if it exists
+            struct_type_enum_set = False
+            members = typeinfo.elem.findall('.//member')
+            for member in members:
+                membername = noneStr(member.find('name').text)
+
+                # We only care about structures with an sType, which can be included in a pNext chain.
+                if membername == self.getStructTypeVarName():
+                    # Check for value in the XML element.
+                    values = member.attrib.get('values')
+                    if values:
+                        self.struct_type_names[typename] = values
```

Comment:
Can't extensions add valid values for the Xr*Base headers?

Reply:
Sure, but I'm not sure what you're point is here.  The parent would be added as a normal struct and then it would find it's children info in the `parentstruct` section below.  Are you saying that parents shouldn't have a type?  That's true and we could optimize that in the future by skipping this if that field exists.

Reply:
TL;DR Never mind.

I was thinking "values" were the valid set of values for the member, and that for something like `XrCompositionLayerBaseHeader` the `values` member would list all the known values.

I was wrong.  `XrCompositionLayerBaseHeader` has *no* values field, and all values fields for `XrStructureType` have only one value.